oresat_dxwifi/camera/camera.py

```diff
@@ -1,107 +1,234 @@
-    parser.add_argument("-del", "--delete-frames", default = False, action = "store_true",
-                        help = "Delete the raw frames after completion.")
+    parser = argparse.ArgumentParser(description="Capture and encode low-fps videos.")
+    parser.add_argument(
+        "-bin",
+        "--binary",
+        default="./build/capture",
+        help="Path to imagecapture binary.",
+    )
+    parser.add_argument(
+        "-dev",
+        "--device",
+        default="/dev/v4l/by-id/usb-Empia._UVC_Video_Device_12345678901234567890-video-index0",
+        help="Path to video device.",
+    )
+    parser.add_argument(
+        "-vx",
+        "--video-x",
+        type=int,
+        default=640,
+        help="Horizontal resolution of video.",
+    )
+    parser.add_argument(
+        "-vy", "--video-y", type=int, default=480, help="Vertical resolution of video."
+    )
+    parser.add_argument(
+        "-fps",
+        "--frames-per-second",
+        dest="fps",
+        type=int,
+        default=4,
+        help="Desired frames per second. (Note: this will probably only work at or under 10fps.)",
+    )
+    parser.add_argument(
+        "-spv",
+        "--seconds-per-video",
+        dest="spv",
+        type=int,
+        default=3,
+        help="Seconds per created video. Must divide total duration.",
+    )
+    parser.add_argument(
+        "-br",
+        "--bit-rate",
+        type=int,
+        default=100,
+        help="Bit rate of H.264 encoded videos.",
+    )
+    parser.add_argument(
+        "-t",
+        "--total-duration",
+        type=int,
+        required=True,
+        help="Total capture duration, in seconds.",
+    )
+    parser.add_argument(
+        "-io",
+        "--image-output",
+        required=True,
+        help="Directory where raw frames should be stored. (Will create / overwrite.)",
+    )
+    parser.add_argument(
+        "-vo",
+        "--video-output",
+        required=True,
+        help="Directory where output videos should be stored. (Will create / overwrite.)",
+    )
+    parser.add_argument(
+        "-del",
+        "--delete-frames",
+        default=False,
+        action="store_true",
+        help="Delete the raw frames after completion.",
+    )
+
+    return parser
+
+
+class CameraV4l2Interface:
+    def __init__(
+        self,
+        binary,
+        device,
+        video_x,
+        video_y,
+        fps,
+        spv,
+        br,
+        duration,
+        image_output,
+        video_output,
+        frames_del,
+    ):
+        self.camera_binary_path = binary
+        self.device_path = device
+        self.horizontal_video_resolution = video_x
+        self.vertical_video_resolution = video_y
+        self.frames_per_second = fps
+        self.seconds_per_video = spv
+        self.bit_rate = br
+        self.total_duration = duration
+        self.image_output_directory_path = image_output
```

Comment:
`image_output_directory` or `image_output_path`

Likewise for video output below